src/update-utmp/update-utmp.c

```diff
@@ -97,8 +97,15 @@ static int get_current_runlevel(Context *c) {
                                 "ActiveState",
                                 &error,
                                 &state);
+                if (sd_bus_error_has_names(&error,
+                                           SD_BUS_ERROR_NO_REPLY,
+                                           SD_BUS_ERROR_DISCONNECTED)) {
+                        /* systemd might have dropped off momentarily, let's not make this an error. */
+                        log_debug_errno(r, "Failed to get state of %s, ignoring: %s", table[i].special, bus_error_message(&error, r));
+                        return 0;
+                }
```

Comment:
Hmm, a better strategy would be to disconnect and try the whole thing anew on a new bus connection. (up to some limit). That way we can pretty reliable ensure that utmp will eventually be updated.